### PR TITLE
Modifies pre-commit checks. -- SGPROJ-1600

### PR DESCRIPTION
**Pull request by DJKaswa**
## Change description

> Removed unnecessary files, and adds tfenv to track terraform version and install it automatically with tfenv.

## Type of change
- [ ] Bug fix (fixes an issue)
- [x] New feature (adds functionality)

## Related issues

> Fix [#1]() 

### Development

- [x] Lint rules pass locally
- [ ] Application changes have been tested thoroughly
- [x] Automated tests covering modified code pass

### Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### Code review 

- [ ] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [x] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [x] Pull request linked to task tracker where applicable

### CODE REVIEW

File: .github/workflows/pre-commit.yml [hunk 1]
```diff
@@ -4,51 +4,6 @@ on:
...
-      - name: Inits
-        run: |
-           tflint --init
-      - name: Execute pre-commit
-        # Run all pre-commit checks on min version supported
-        if: ${{ matrix.version ==  needs.getBaseVersion.outputs.minVersion }}
-        run: pre-commit run --color=always --show-diff-on-failure --all-files
+  build:
+    uses: StratusGrid/workflow-config/.github/workflows/pre-commit.yml@main
+    secrets: inherit
```
Comment (JoseMotto-sg) [inline .github/workflows/pre-commit.yml]: Delete this
```suggestion
    secrets: inherit
```

File: .terraform-version [hunk 2]
```diff
@@ -0,0 +1 @@
+1.2.8
```
Comment (JoseMotto-sg) [inline .terraform-version:1]:
```suggestion
1.3.0
```

**Review by JoseMotto-sg — CHANGES_REQUESTED** [review]

**Comment by DJKaswa** [comment]
@JoseMotto-sg please review my response to your suggestions/changes requested. Other than that, we should be ready to merge this.

**Review by JoseMotto-sg — APPROVED** [review]
LGTM

**Pull request merged**